### ISSUE TITLE
Latency

### ISSUE THREAD

**Issue by krintek** [stonith404/pingvin-share#98]
![image](https://user-images.githubusercontent.com/87629857/218260970-88ab1454-c35d-48f6-b789-672d4a0ab5ed.png)

I use nginx reverse proxy to handle my traffics. Every service I use has fairly low latency but unfortunately Pingvin Share has latency above 1800 ms and I thought whether if anyone could fix it  :/

BTW I LOVE PINGVIN <3

**Comment by stonith404** [comment]
Thanks for your report :) I can't reproduce this, I have a good latency:
<img width="966" alt="image" src="https://user-images.githubusercontent.com/58886915/218265523-32ce577a-14df-4aea-965a-83e81daf1ddb.png">

How does your monitor tool measure the latency? With a HTTP request, ping or something else? And which version of Pingvin Share are you using?

**Comment by krintek** [comment]
I measure the latency with a HTTP request! I'm using the latest version btw.

**Comment by stonith404** [comment]
It's difficult to find out exactly what the problem is... I have a response time between 150ms and 300ms.
Do you have the same latency with the following command?
```
curl -o /dev/null -s -w 'Total: %{time_total}s\n' https://yourinstance.com
```

**Comment by krintek** [comment]
![image](https://user-images.githubusercontent.com/87629857/218348654-e037c284-5393-4f26-8542-9988d3e37bf7.png)
 Mine is wayyyyyy higher

**Comment by Hukuma1** [comment]
One more data point:
```
Total: 0.075364s
Total: 0.050012s
Total: 0.074251s
```

**Comment by stonith404** [comment]
@krintek It's really strange. Could you execute the same command again but without the proxy in front? This can be done when you curl the website directly on your server e.g `curl -o /dev/null -s -w 'Total: %{time_total}s\n' http://localhost:3000`

**Comment by jl-678** [comment]
The screenshot from @Hukuma1 seems to come from something like Heimdall or Homepage. I use Homepage and see longer ping times too but far shorter than shown by the OP. (As an aside, Homepage seems to show longer ping times for all services for some reason.)

The screenshot from @stonith404 is from Uptime Kuma which I run too. Adding Pingvin to it shows much lower ping times which are in line with most other Docker services in my environment.

In short, I cannot reproduce this issue either and wonder whether it could be due to how Heimdall/Homepage calculate and report on ping.

**Comment by Hukuma1** [comment]
To clarify, my output is just copy/paste of my results from terminal in code brackets on Github. That is what I got when using [@stonith404's command:](https://github.com/stonith404/pingvin-share/issues/98#issuecomment-1427135949) `curl -o /dev/null -s -w 'Total: %{time_total}s\n' https://yourinstance.com`

**Comment by jl-678** [comment]
Okay, just to be clear I have pretty much the same config as you including using Nginx Proxy Manager. (I am also running Uptime Kuma and Homepage.) Here is the data that I see:

```
curl -o /dev/null -s -w 'Total: %{time_total}s\n' https://public_url.com <- Public URL passing through NPM
Total: 0.247392s
Total: 0.210484s
Total: 0.218479s
Total: 0.211005s
```

```
curl -o /dev/null -s -w 'Total: %{time_total}s\n' http://<internal IP>:port <- Going straight to Pingvin on the docker host
Total: 0.033462s
Total: 0.036483s
Total: 0.030198s
Total: 0.032151s
```

Out of curiosity, I used the same command locally on the docker host for other containers with web GUIs and found that those are a bit quicker responding between .001 and .008 depending on the app. These are a bit faster, but I am not sure that I would consider the difference meaningful.

**Comment by stonith404** [comment]
Closing this because the creator of the issue didn't answer. Feel free to reopen the issue if it still occurs.
